sklearn/inspection/_plot/tests/test_boundary_decision_display.py

```diff
@@ -485,6 +485,23 @@ def test_dataframe_support(pyplot):
         DecisionBoundaryDisplay.from_estimator(estimator, df, response_method="predict")
 
 
+def test_polars_dataframe_support(pyplot):
+    """Check that passing a polars dataframe at fit and to the Display does not
+    raise warnings.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/28717
+    """
+    pl = pytest.importorskip("polars")
+    df = pl.DataFrame({"col_x": X[:, 0], "col_y": X[:, 1]})
+    estimator = LogisticRegression().fit(df, y)
+
+    with warnings.catch_warnings():
+        # no warnings linked to feature names validation should be raised
+        warnings.simplefilter("error", UserWarning)
+        DecisionBoundaryDisplay.from_estimator(estimator, df, response_method="predict")
+
```

Comment:
We can parametrize the previous test for pandas and polars dataframes. Something like

```py
@pytest.mark.parametrize("constructor_name", ["pandas", "polars"])
def test_dataframe_support(pyplot, constructor_name):
    df = _convert_container(X, constructor_name=constructor_name, column_name=["col_x", "col_y"])
```

Reply:
Done.

Reply:
I just pushed a commit to use ``_convert_container`` which is the preferred way when dealing with different types of containers in tests.